src/test/java/org/cactoos/iterator/SyncIteratorTest.java

```diff
@@ -73,4 +78,209 @@ public void syncIteratorReturnsCorrectValuesWithInternalLock() {
+            final Runnable second = () -> {
+                sync.add(iterator.next());
+            };
+            final Runnable third = () -> {
+                sync.add(iterator.hasNext());
+            };
+            final List<Runnable> test = new ArrayList<>(list.size() + 1);
+            test.add(first);
+            test.add(second);
+            test.add(third);
+            new Concurrent(test).launch();
+            MatcherAssert.assertThat(
+                "Missing the list items(s) (next()).",
+                sync,
+                Matchers.allOf(
+                    Matchers.hasItem("a"),
+                    Matchers.hasItem("b")
+                )
+            );
+            MatcherAssert.assertThat(
+                "Missing hasNext() value.",
+                sync,
+                Matchers.anyOf(
+                    Matchers.hasItem(true),
+                    Matchers.hasItem(false)
+                )
+            );
+        }
+    }
+
+    @Test
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public void correctValuesForConcurrentHasNextHasNext()
+        throws InterruptedException {
+        for (int iter = 0; iter < 5000; iter += 1) {
+            final List<String> list = Arrays.asList("a", "b");
+            final SyncIterator<String> iterator = new SyncIterator<>(
+                list.iterator()
+            );
+            final List<Object> sync =
+                Collections.synchronizedList(
+                    new ArrayList<>(list.size())
+                );
+            final Runnable first = () -> {
+                sync.add(iterator.hasNext());
+            };
+            final Runnable second = () -> {
+                sync.add(iterator.hasNext());
+            };
+            final List<Runnable> test = new ArrayList<>(list.size());
+            test.add(first);
+            test.add(second);
+            new Concurrent(test).launch();
+            MatcherAssert.assertThat(
+                "Missing hasNext() value(s).",
+                sync,
+                Matchers.contains(true, true)
+            );
+        }
+    }
+
+    /**
+     * Tests runnables for concurrency issues.
+     */
+    private final class Concurrent {
+
+        /**
+         * Runnables to run in different threads.
+         */
+        private final List<? extends Runnable> runnables;
+
+        /**
+         * Collected exceptions.
+         */
+        private final List<Throwable> exceptions;
+
+        /**
+         * Thread pool.
+         */
+        private final ExecutorService pool;
+
+        /**
+         * All executor threads are ready.
+         */
+        private final CountDownLatch ready;
+
+        /**
+         * Start countdown with first thread.
+         */
+        private final CountDownLatch init;
+
+        /**
+         * All threads ready.
+         */
+        private final CountDownLatch done;
+
+        Concurrent(final List<? extends Runnable> runnables) {
```

Comment:
@driver733 I think here, instead of a List, you could accept ``Runnable...`` and then in the tests you won't have to instantiate Runnables, add them to a List and use that List. You can directly instantiate ``Concurrent`` and give the Runnables diractly as lambdas.

Reply:
@amihaiemil I agree with you. I will make this change now.